core/src/main/java/org/geysermc/geyser/util/EntityUtils.java

```diff
@@ -223,7 +223,10 @@ public static void updateMountOffset(Entity passenger, Entity mount, boolean rid
                 yOffset -= armorStand.getYOffset();
             }
             Vector3f offset = Vector3f.from(xOffset, yOffset, zOffset);
-            passenger.setRiderSeatPosition(offset);
+
+            passenger.setRiderSeatPosition( passenger instanceof TextDisplayEntity textDisplay ?
+                    textDisplay.baseTranslation.add(offset).add(0f, 0.2f, 0f) : offset
+            );
```

Comment:
As above - we shouldn't add specific entity workaround to general utility methods. Instead, please find a way to either move this to the text display entity,  or add it to a new switch case above.